apps/prairielearn/src/migrations/20250204232139_course_instance_usages__create.sql

```diff
@@ -0,0 +1,31 @@
+CREATE TYPE enum_course_instance_usages_type AS ENUM(
+  -- The user made at least one submission.
+  'Submission',
+  -- Total compute time in external grading.
+  'External grading',
+  -- Total compute time in workspaces.
+  'Workspace'
+);
+
+CREATE TABLE course_instance_usages (
+  id BIGSERIAL PRIMARY KEY,
+  type enum_course_instance_usages_type NOT NULL,
+  institution_id BIGINT NOT NULL REFERENCES institutions (id),
+  course_id BIGINT NOT NULL REFERENCES pl_courses (id),
+  course_instance_id BIGINT REFERENCES course_instances (id),
```

Comment:
Note that `course_instance_id` may be NULL in the case that the usage is for course staff operating outside of a course instance.

If a course staff member is sometimes inside a course instance and sometimes outside, we'll record two distinct rows for them, one with `(course_id, course_instance_id)` and one with `(course_id, NULL)`. This is ok because any given usage will only produce one type of row or the other, so we won't double-count, and all of the analytics queries are aggregating over both types of rows.

Reply:
Done in 37277fa3986d63d714b516114050b9c8126cc927